src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -706,7 +667,7 @@ private IOperation CreateBoundConversionOperation(BoundConversion boundConversio
                 bool isChecked = conversion.IsNumeric && boundConversion.Checked;
                 ITypeSymbol type = boundConversion.Type;
                 Optional<object> constantValue = ConvertToOptional(boundConversion.ConstantValue);
-                return new LazyCSharpConversionExpression(operand, conversion, isExplicit, isTryCast, isChecked, _semanticModel, syntax, type, constantValue, isImplicit);
+                return new LazyCSharpConversionExpression(operand, conversion, isExplicit, isTryCast, isChecked, _semanticModel, syntax, type, constantValue, isImplicit || !isExplicit);
```

Comment:
implicit conversion require checking both compiler generated + explicit cast in code since compiler won't mark it as compiler generated